File: olds/recorrido/google-maps-routes_v1/python/src/route_finder.py
#!/usr/bin/env python3
import googlemaps
import sys
from datetime import datetime

class GoogleMapsRouteFinder:
    def __init__(self, api_key):
        self.gmaps = googlemaps.Client(key=api_key)

    def find_route(self, origin, destination, mode="driving"):
        try:
            # Obtener direcciones
            directions = self.gmaps.directions(
                origin,
                destination,
                mode=mode,
                departure_time=datetime.now()
            )

            if not directions:
                print("❌ No se encontraron rutas")
                return

            route = directions[0]
            leg = route['legs'][0]

            print("\n✅ Ruta encontrada!\n")
            print(f"📍 Origen: {leg['start_address']}")
            print(f"🎯 Destino: {leg['end_address']}")
            print(f"📏 Distancia: {leg['distance']['text']}")
            print(f"⏱️  Duración: {leg['duration']['text']}")

            print("\n📝 Instrucciones:")
            for i, step in enumerate(leg['steps'], 1):
                instruction = step['html_instructions'].replace('<b>', '').replace('</b>', '')
                print(f"{i}. {instruction}")
                print(f"   {step['distance']['text']} ({step['duration']['text']})\n")

        except Exception as e:
            print(f"❌ Error: {e}")

    def get_route_polyline(self, origin, destination, mode="driving"):
        """Obtiene el polyline de la ruta para dibujar"""
        directions = self.gmaps.directions(origin, destination, mode=mode)
        if directions:
            return directions[0]['overview_polyline']['points']
        return None

def main():
    API_KEY = "AIzaSyCGbLfwvbxGV_bWvbtkHQrV8hHM7flIwMo"

    print("=== Google Maps Route Finder (Python) ===")
    origin = input("Origen: ")
    destination = input("Destino: ")

    print("\nModos de viaje:")
    print("1. Coche")
    print("2. A pie")
    print("3. Bicicleta")
    print("4. Transporte público")
    choice = input("Elige una opción (1-4): ")

    mode_map = {
        '1': 'driving',
        '2': 'walking',
        '3': 'bicycling',
        '4': 'transit'
    }
    mode = mode_map.get(choice, 'driving')

    finder = GoogleMapsRouteFinder(API_KEY)
    finder.find_route(origin, destination, mode)

if __name__ == "__main__":
    main()

File: olds/recorrido/google-maps-routes_v1/java/src/RouteFinder.java
import com.google.maps.DirectionsApi;
import com.google.maps.GeoApiContext;
import com.google.maps.model.DirectionsResult;
import com.google.maps.model.TravelMode;
import java.util.Scanner;

public class RouteFinder {
    private static final String API_KEY = "AIzaSyCGbLfwvbxGV_bWvbtkHQrV8hHM7flIwMo";

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("=== Google Maps Route Finder ===");
        System.out.print("Origen: ");
        String origin = scanner.nextLine();
        System.out.print("Destino: ");
        String destination = scanner.nextLine();

        // Menú de modo de viaje
        System.out.println("\nModos de viaje:");
        System.out.println("1. Coche");
        System.out.println("2. A pie");
        System.out.println("3. Bicicleta");
        System.out.println("4. Transporte público");
        System.out.print("Elige una opción (1-4): ");
        int modeChoice = scanner.nextInt();

        TravelMode mode;
        switch(modeChoice) {
            case 1: mode = TravelMode.DRIVING; break;
            case 2: mode = TravelMode.WALKING; break;
            case 3: mode = TravelMode.BICYCLING; break;
            case 4: mode = TravelMode.TRANSIT; break;
            default: mode = TravelMode.DRIVING;
        }

        findRoute(origin, destination, mode);
    }

    private static void findRoute(String origin, String destination, TravelMode mode) {
        try {
            GeoApiContext context = new GeoApiContext.Builder()
                .apiKey(API_KEY)
                .build();

            DirectionsResult result = DirectionsApi.newRequest(context)
                .origin(origin)
                .destination(destination)
                .mode(mode)
                .await();

            if (result.routes.length > 0) {
                var route = result.routes[0];
                var leg = route.legs[0];

                System.out.println("\n✅ Ruta encontrada!\n");
                System.out.println("📍 Origen: " + leg.startAddress);
                System.out.println("🎯 Destino: " + leg.endAddress);
                System.out.println("📏 Distancia: " + leg.distance.humanReadable);
                System.out.println("⏱️  Duración: " + leg.duration.humanReadable);
                System.out.println("\n📝 Instrucciones:");

                int stepNum = 1;
                for (var step : leg.steps) {
                    String instruction = step.htmlInstructions.replaceAll("<[^>]*>", "");
                    System.out.println(stepNum++ + ". " + instruction);
                    System.out.println("   " + step.distance.humanReadable + " (" + step.duration.humanReadable + ")\n");
                }
            } else {
                System.out.println("❌ No se encontraron rutas");
            }

            context.shutdown();
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}
